File: Behavioral/Memento/memento.cs
class Memento
{
    public string State { get; }
    public Memento(string state) => State = state;
}

class Editor
{
    public string Text = "";

    public Memento Save() => new Memento(Text);
    public void Restore(Memento m) => Text = m.State;
}

File: Behavioral/Memento/memento.py
class Memento:
    def __init__(self, state):
        self.state = state

class Editor:
    def __init__(self):
        self.text = ""

    def save(self):
        return Memento(self.text)

    def restore(self, memento):
        self.text = memento.state
